mitiq/zne/scaling/layer_scaling.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2023 Unitary Fund
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Functions for layer-wise unitary folding on supported circuits."""
+from typing import List
+from cirq import Circuit, inverse, Moment
```

Comment:
To reduce typing ambiguity what do you think about something like this?
```suggestion
import cirq
from cirq import inverse, Moment
```
This way in type hints we will use `cirq.Circuit`, as other frontends can call their abstraction `Circuit` as well. Pretty minor, however. WDYT?